OutboxSample/Infrastructure/DataAccess/IDbCommandExtensions.cs

```diff
@@ -36,4 +36,80 @@ public static DbParameter CreateParameter(this IDbCommand command, string parame
 
         return parameter;
     }
+
+    public static Task<int> ExecuteNonQueryAsync(this IDbCommand command)
+    {
+        ArgumentNullException.ThrowIfNull(command, nameof(command));
+
+        command = command.GetLiveCommand();
+
+        if (command is DbCommand dbCommand)
+        {
+            return dbCommand.ExecuteNonQueryAsync();
```

Comment:
It doesn't look very intuitive, but we need to consider whether exception propagation difference between this:
```csharp
return dbCommand.ExecuteNonQueryAsync();
```
and this:
```csharp
return await dbCommand.ExecuteNonQueryAsync();
```
matters. By default,  might make sense to go with `await` even considering the extra overhead of the async-state machine creation.
More details: https://stackoverflow.com/questions/19098143/what-is-the-purpose-of-return-await-in-c/19098209#19098209